src/Compilers/VisualBasic/Portable/Semantics/OverloadResolution.vb

```diff
@@ -2703,8 +2703,14 @@ Done:
 
             For i As Integer = 0 To arguments.Length - 1 Step 1
 
-                If Not argumentNames.IsDefault AndAlso argumentNames(i) IsNot Nothing Then
-                    ' First named argument
+                If Not argumentNames.IsDefault AndAlso argumentNames(i) IsNot Nothing AndAlso
+                    NamedArgumentMustBeTrailing(candidate.Candidate, argumentNames(i), paramIndex) Then
```

Comment:
>paramIndex [](start = 87, length = 10)

I think this should be ```i``` instead, paramIndex doesn't always match argument index.